src/connection.js

```diff
@@ -599,6 +633,45 @@ class Connection extends EventEmitter {
       this.emit('charsetChange', token.newValue);
     });
 
+    this.tokenStreamParser.on('fedAuthInfo', (token) => {
+      const clientId = '7f98cb04-cd1e-40df-9140-3bf7e2cea4db';
+      if (token.fedAuthInfoData.stsurl && token.fedAuthInfoData.spn) {
+        this.fedAuthInfo.responsePending = true;
+        var context = new AuthenticationContext(token.fedAuthInfoData.stsurl);
+        context.acquireTokenWithUsernamePassword(token.fedAuthInfoData.spn, this.config.userName, this.config.password, clientId, (err, tokenResponse) => {
+          if (err) {
+            this.fedAuthInfo.responsePending = false;
+            //console.log('error');
+            this.loginError = ConnectionError('Security token could not be authenticated or authorized.', 'EFEDAUTH');
+          } else {
+            this.fedAuthInfo.responsePending = false;
+            this.fedAuthInfo.token = tokenResponse;
+          }
+        });
+      }
+    });
+
+    this.tokenStreamParser.on('featureExtAck', (token) => {
+      let value = undefined;
```

Comment:
Right now there is no check to verify if the server has sent `featureExtAck`. Connection state switches from `SENT_LOGIN7_WITH_FEDAUTH` to `SENT_LOGIN7_WITH_STANDARD_LOGIN` after sending the `FEDAUTH_TOKEN` packet, so the implementation uses the general check in [`processLogin7NTLMResponse`](https://github.com/tediousjs/tedious/pull/707/files#diff-875da84119066ec244742e95d1209f0cL1181) to switch to `LOGGED_IN` state irrespective of Server's lack of response for `FEDAUTH_TOKEN`. Can you take care of this? It can go in `Sent LOGIN7 Record with Complete Authentication Token State` (TDS spec 3.2.5.3)